pkg/controller/watch/resource_watcher.go

```diff
@@ -1,17 +1,72 @@
 package watch
 
 import (
+	"go.uber.org/zap"
 	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func NewResourceWatcher() ResourceWatcher {
 	return ResourceWatcher{
-		WatchedResources: map[WatchedObject][]types.NamespacedName{},
+		WatchedResources: map[WatchedObject][]client.ObjectKey{},
 	}
 }
 
+// ResourceWatcher is the object containing the map of watched_resource -> []dependant_resource.
 type ResourceWatcher struct {
-	WatchedResources map[WatchedObject][]types.NamespacedName
+	WatchedResources map[WatchedObject][]client.ObjectKey
```

Comment:
What do you think about converting this to a set? Logically, that's what it currently looks like anyway with `pos()` and `remove()` and the whole keeping values unique business...

Reply:
I think you mean the `map[client.ObjectKey]bool`?
yes, this will work out, will make the code more concise 👍